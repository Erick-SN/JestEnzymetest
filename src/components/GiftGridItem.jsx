import React from 'react';
const GiftGridItem = ({ title, url }) => {
  return (
    <div className='card animate_animated animate_fadeIn'>
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

export default GiftGridItem;
