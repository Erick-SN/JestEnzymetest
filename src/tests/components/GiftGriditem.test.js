import React from 'react';
import GiftGridItem from '../../components/GiftGridItem';
import { shallow } from 'enzyme';
describe('Test at <GiftGridItem />', () => {
  const title = 'image';
  const url = 'https://image.com';
  const wrapper = shallow(<GiftGridItem title={title} url={url} />);
  test('Component render', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('title at p', () => {
    const p = wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });
  test('Image must be equal to URL', () => {
    const image = wrapper.find('img');
    expect(image.prop('src')).toBe(url);
    expect(image.prop('alt')).toBe(title);
  });
});
