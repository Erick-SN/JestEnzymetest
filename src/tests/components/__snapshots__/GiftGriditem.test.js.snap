// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test at <GiftGridItem /> Component render 1`] = `ShallowWrapper {}`;
